File: 2572-append-characters-to-string-to-make-subsequence/2572-append-characters-to-string-to-make-subsequence.py
class Solution:
    def appendCharacters(self, s: str, t: str) -> int:
        s_left=0
        t_left=0
        while s_left<len(s) and t_left<len(t):
            if s[s_left]==t[t_left]:
                t_left+=1
            s_left+=1
        return len(t)-t_left

        

      

       



        

File: 2572-append-characters-to-string-to-make-subsequence/2572-append-characters-to-string-to-make-subsequence.java
class Solution {
    public int appendCharacters(String s, String t) {
        int s_left=0,t_left=0;
        int s_length=s.length(),t_length=t.length();

        while (s_left<s_length && t_left<t_length){
            if(s.charAt(s_left)==t.charAt(t_left)){
                t_left++;
            }
            s_left++;
        }
        return t_length-t_left;
        
    }
}